src/components/ui/Publishers.jsx

```diff
@@ -0,0 +1,101 @@
+import classNames from "classnames";
+import { useEffect, useState } from "react";
+import { BiCategoryAlt } from "react-icons/bi";
+import { BsGlobe } from "react-icons/bs";
+import { IoLanguage } from "react-icons/io5";
+import { useDispatch, useSelector } from "react-redux";
+import { useFetchPublishersQuery } from "../../features/api/apiSlice";
+import { selectPublisher } from "../../features/articles/articlesSlice";
+import { getCountryFromCode, getLanguageFromCode } from "../../helpers/articles";
+import PublisherRowSkeleton from "../skeletons/PublisherRow";
+
+const Publishers = () => {
+  const dispatch = useDispatch();
+  const { selectedPublisher } = useSelector((state) => state.articles);
+  const { data, isFetching } = useFetchPublishersQuery();
+  const [visiblePublishers, setVisiblePublishers] = useState(data);
+
+  const handleSelectPublisher = (publisher) => dispatch(selectPublisher(publisher));
+  const resetPublishers = (event) => {
+    event.stopPropagation();
+    dispatch(selectPublisher(""));
+  };
+
+  const badgeClasses = classNames(
+    "px-2 capitalize flex items-center border rounded-full gap-[2px] md:py-[1px] border-gray-300 w-18 sm:w-auto"
+  );
+  const setPublisherWrapperClasses = (publisher) =>
+    classNames("pb-2 pt-1 md:pt-2 md:pb-3 px-2 cursor-pointer text-slate-600 hover:bg-[#f1eee8]", {
+      "bg-[#eae6df] sticky w-full top-0 bottom-0": publisher.id === selectedPublisher?.id,
+    });
+
+  const handleSearch = (event) => {
+    const { value } = event.target;
+    setVisiblePublishers(
+      data.filter((publisher) => publisher.name.toLowerCase().includes(value.toLowerCase()))
+    );
+  };
+
+  useEffect(() => {
+    if (data) setVisiblePublishers(data);
+  }, [data]);
+
+  return (
+    <div className="flex flex-col h-full">
+      <input
+        type="text"
+        onChange={handleSearch}
+        placeholder="Search for publisher..."
+        className="px-2 text-sm border border-gray-400 rounded outline-none h-fit py-[6px] placeholder:text-gray-400"
+      />
+      <div className="w-full h-full mt-2 overflow-hidden rounded-lg shadow md:border">
+        <div className="divide-y-2 divide-zinc-300 overflow-y-auto h-full max-h-[600px] relative scrollbar-thumb-[#8d6a43] hover:scrollbar-thumb-[#725738] scrollbar-track-[#EBE1D7] scrollbar-thin scrollbar-thumb-rounded-full scrollbar-w-8">
+          {isFetching &&
+            Array(8)
+              .fill(0)
+              .map((_item, index) => <PublisherRowSkeleton key={index} />)}
+
+          {!isFetching &&
+            visiblePublishers &&
+            visiblePublishers.map((publisher) => (
```

Comment:
It will be better to use a ternary operator than checking whether isFetching is true and then if it's not again.
```suggestion
          {isFetching?
            Array(8)
              .fill(0)
              .map((_item, index) => <PublisherRowSkeleton key={index} />)
            : visiblePublishers?.map((publisher) => (
```